openssl-fips/plan.sh

```diff
@@ -0,0 +1,50 @@
+pkg_name=openssl-fips
+pkg_origin=core
+pkg_version=2.0.16
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_description="\
+The OpenSSL FIPS Object Module v2.0 provide cryptographic services to \
+external applications. The FIPS Object Module provides an API for invocation \
+of FIPS approved cryptographic functions from calling applications, and is \
+designed for use in conjunction with standard OpenSSL 1.0.1 and 1.0.2 \
+distributions. \
+"
+pkg_upstream_url="https://www.openssl.org"
+pkg_license=('OpenSSL')
+pkg_source="https://www.openssl.org/source/${pkg_name}-${pkg_version}.tar.gz"
+pkg_shasum="a3cd13d0521d22dd939063d3b4a0d4ce24494374b91408a05bdaca8b681c63d4"
+pkg_filename="${pkg_name}-${pkg_version}.tar.gz"
+pkg_deps=(
+  core/glibc
+)
+pkg_build_deps=(
+  core/make
+  core/perl
+  core/gcc
+)
+pkg_bin_dirs=(bin)
+pkg_include_dirs=(include)
+pkg_lib_dirs=(lib)
+
+do_prepare() {
+  do_default_prepare
+  pushd "${HAB_CACHE_SRC_PATH}/${pkg_name}-${pkg_version}"
+  # Purge the codebase of the hardcoded reliance on `/bin/rm`.
+  grep -lr '/bin/rm' . | while read -r f; do
```

Comment:
Does this violate https://wiki.openssl.org/index.php/FIPS_Build_Guidelines - in particular, it modifies the contents of the tarball. Instead, you may have to break integrity. 

```bash
if [[ ! -f "/bin/rm" ]]; then
   # link to an rm at /bin/rm, since one doesn't exist.
fi
```

Any other move might invalidate your build of the FIPS container, yes?

Reply:
hm, @echohack can you remind me why we added this in?    I just did a test build with that code removed and it seems to build just fine.

Reply:
Specifically i did this because the compile of the canister expects that /bin/rm is hard coded and exists on the operating system.

I think there should be a better way of doing this (and I'm happy to sit down and do it).

Reply:
My point is, by my reading of the wiki link above, doing it means you're violating the sanctity of the build process, which is the reason you want the FIPS cannister in the first place. So you have to leave the tarball as it came, even with `/bin/rm` hard coded. So check for the existence of `/bin/rm`, and if it's not there, go ahead and link it if we're in the studio. It'll get destroyed when the studio exits anyway, and if you try and build outside the studio, just abort the build.

Reply:
You're totally right. Sorry for the lack of response, I was on pretty shady wifi until now!

Yes that's exactly the plan. Once I am back from conference I want to sit down and fix this issue!

In addition, I'd like to spend some time reading over the whole spec in detail, because all we've done to this point is simply prove that "it works", and I'd like to be more thorough than that.